app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,14 @@
 class Api::BaseController < ApplicationController
   layout nil
+
+  # Method to log login attempts
+  def log_login_attempt(user_id:, attempted_at:, successful:, ip_address:)
```

Comment:
Consider using Rails' built-in request object to get the 'ip_address' instead of passing it as a parameter, unless there is a specific reason to do otherwise. This can help ensure the IP address is accurate and not manipulated.